examples/gno.land/r/demo/groups/public.gno

```diff
@@ -0,0 +1,94 @@
+package groups
+
+import (
+	"std"
+
+	"gno.land/r/demo/users"
+)
+
+//----------------------------------------
+// Public facing functions
+
+func GetGroupIDFromName(name string) (GroupID, bool) {
+
+	groupI, exists := gGroupsByName.Get(name)
+	if !exists {
+		return 0, false
+	}
+	return groupI.(*Group).id, true
+}
+
+func CreateGroup(name string) GroupID {
+
+	std.AssertOriginCall()
+	gid := incGetGroupID()
+	caller := std.GetOrigCaller()
+	if usernameOf(caller) == "" {
+		panic("unauthorized")
+	}
```

Comment:
Shouldn't this check be moved towards the top of the method, before you globally modify the `gid`?

Reply:
Also, I think `usernameOf()` should panic instead?

Reply:
added panic at usernameOf() method